scipy/interpolate/_bsplines.py

```diff
@@ -314,6 +322,14 @@ def __call__(self, x, nu=0, extrapolate=None):
         x = np.asarray(x)
         x_shape, x_ndim = x.shape, x.ndim
         x = np.ascontiguousarray(x.ravel(), dtype=np.float_)
+
+        # With periodic extrapolation we map x to the segment
+        # (self.t[k], self.t[-k]).
+        if extrapolate == 'periodic':
```

Comment:
While admittedly not expert in B-splines, my understanding is that 

a) The interval can be considered closed, because there is virtually no open intervals in numerical methods. Also refer to Notes in the docstring.
b) Whether it mathematically open or closed doesn't change the fact that the initial point is `t[k]` and the length is `t[-k] - t[k]`.